### DISCUSSION TITLE
Discussion: event subscription in object initializers

### DISCUSSION THREAD

**Discussion by alrz** [dotnet/csharplang#307] [Language Ideas]
Moved from https://github.com/dotnet/roslyn/issues/5365

Extend object initializer block to support event subscriptions:
```cs
var timer = new DispatcherTimer {
    Tick += delegate {},
    Interval = TimeSpan.FromSeconds(1d),
};
```
->
```cs
var timer = new DispatcherTimer();
timer.Tick += delegate {};
timer.Interval = TimeSpan.FromSeconds(1d);
```
**Open question:** should we allow multiple event subscriptions for a single event at a time?
```cs
var timer = new C{
    Event += delegate {},
    Event += delegate {},
};
```

**Comment by drewnoakes** [comment]
I'd love to see this. It'd allow several construction/initialisation operations to become single expressions, which play nicely with other C# features that reduce nesting such as expression bodies, null coalescing, ternary conditionals...
 
There's some support for this idea [on StackOverflow too](https://stackoverflow.com/q/3993601/24874).

**Comment by CortiWins** [comment]
I would love this!

**Comment by blair-ahlquist** [comment]
No. The feature that should be implemented is https://github.com/dotnet/csharplang/issues/1652. It covers the requested feature and does so much more.

**Comment by AlmightyFuzz** [comment]
I would really like to see this feature added. It would tidy up a lot of existing initialization code where you have to do the following following if you want to use Object Initialization:

    var timer = new Timer()
    {
      Interval = TimeSpan.FromSeconds(5),
      // further property settings
    };
    timer.Tick += OnTick;

I'm curious as to why this wasn't added when the Object Initializer functionality was introduced? It _looks_ like a trivial thing to add.

**Comment by CyrusNajmabadi** [comment]
I'll champion this.

**Reply by markusschaber** [reply]
Any progress?

**Reply by jnm2** [reply]
Yes, this falls into https://github.com/dotnet/csharplang/issues/5176
